corehq/apps/sso/utils/entra.py

```diff
@@ -26,10 +26,17 @@ def get_all_usernames_of_the_idp_from_entra(idp):
     config = _configure_idp(idp)
 
     # Create a preferably long-lived app instance which maintains a token cache.
-    app = msal.ConfidentialClientApplication(
-        config["client_id"], authority=config["authority"],
-        client_credential=config["secret"],
-    )
+    try:
+        app = msal.ConfidentialClientApplication(
+            config["client_id"], authority=config["authority"],
+            client_credential=config["secret"],
+        )
+    except ValueError as e:
+        if "check your tenant name" in str(e):
+            raise EntraVerificationFailed(error="invalid_tenant",
+                                          message="Please double check your tenant id is correct")
```

Comment:
Is there a reason to add our own message here rather than passing along the `ValueError` message that `msal` uses? Maybe this is more in line with other `EntraVerificationFailed` messages we raise.

Reply:
Nevermind, I see in the next commit that we do use our own messages for this.

Reply:
Thanks Evan! I find the long error message they returned is less helpful, based on our code, I can narrow down the reason is tenant id is wrong.